### PR TITLE
Add OpenAPI workflow

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
**Changes**
Add a new workflow called that does the following things:
- Generate OpenAPI spec for master branch
  - although already available via azure on repo.jellyfin.org, this adds it the the GitHub interface
- Generate OpenAPI spec for master branch
  - Useful when I need to test something with the Kotlin SDK, or to inspect changes in the file
- Calculate difference between PR spec and baseline (target branch) spec using [openapi-diff](https://github.com/OpenAPITools/openapi-diff)
  - Diff is rendered using markdown and added as a comment to the PR
  - If the PR is updated it will update the existing comment
  - Comment uses a `<details>` element to prevent skyscraper tall comments
    - For example, when updating BaseItemDto

The comments will help reviewers to determine if a PR contains breaking API changes, which might not be easy to notice in the code reviews.

**To do**
Some work left for this PR that I need some help with:

- Need to use `pull_request_target` instead of `pull_request` BUT this would allow a malicious actor to expose secrets by making changes to the .NET code. No idea how to fix.
  - Need to use jellyfin-bot for comment
- Detect the commit of the base branch the head branch is based on to use as a baseline to prevent changes made in the API in base showing up in the diff.
  - example: property removed from baseitemdto in master would also show up in this PR if the change wasn't in master when this branch was created.
  - I don't think this is a huge issue right now as API changes are scarce, it could become an issue with jf 11 dev though.

**Issues**

**Example**
See my personal fork for an example of this workflow: https://github.com/nielsvanvelzen/jellyfin/pull/91

**Issues**

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
I'm marking the PR as ready now since it does work, the "to do" list contains 2 items that should be considered before merging though.

**Comment by ferferga** [comment]
The ``pull_request_target`` event checkouts code from the base branch instead [(that's what I understand from docs)](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target).

However, for extra peace of mind, why not run this on master commits?

**Comment by nielsvanvelzen** [comment]
> However, for extra peace of mind, why not run this on master commits?

We need the openapi.json file for both the PR branch and the target (=master) branch to compare.

**Review by crobibero — APPROVED** [review]

File: .github/workflows/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+jobs:
+  openapi-head:
+    name: OpenAPI - HEAD
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
```
Comment (cvium) [inline .github/workflows/openapi.yml:16]: Change to
```
    dotnet-version: '6.0.x'
    include-prerelease: true
```

**Review by cvium — CHANGES_REQUESTED** [review]

**Review by cvium — APPROVED** [review]

**Pull request merged**